File: src/client/src/models/invoice/invoiceview.model.ts
import { MasterBaseModel } from "../../core/models/master-base.model";
import { MovieViewModel } from "../movie/movie-view-model";
import { TicketViewModel } from "../ticket/ticketview.model";

export class InvoiceViewModel extends MasterBaseModel{
    public userId!: string;
    public userFullName!: string;
    public userEmail!: string;
    public userPhoneNumber!: string;
    public totalMoney!: number;
    public addedScore!: number;
    public tickets!: TicketViewModel[];
    public roomName!: string;
    public movieName!: string;
    public showDate!: Date;
    public startTime!: string;
    public ticketIssued!: boolean;
    public showTimeId!: string;
    public movieVersion!: number;
}

File: src/server/MovieTheater.Business/ViewModels/Invoice/InvoiceViewModel.cs
using MovieTheater.Business.ViewModels.Ticket;

namespace MovieTheater.Business.ViewModels.Invoice;

public class InvoiceViewModel : MasterBaseViewModel
{
    public required string UserFullName { get; set; }
    public required string UserEmail { get; set; }
    public required string UserPhoneNumber { get; set; }
    public double TotalMoney { get; set; }
    public int AddedScore { get; set; }
    public required string RoomName { get; set; }
    public required string MovieName { get; set; }
    public required DateOnly ShowDate { get; set; }
    public required TimeSpan StartTime { get; set; }
    public required List<TicketViewModel> Tickets { get; set; }
}
